src/app/api/services/paginator/paginator.service.ts

```diff
@@ -0,0 +1,203 @@
+            this.links = {};
+            this._filter = filter;
+        }
+    }
+
+    get sortField(): string{
+        return this._sortField;
+    }
+
+    set sortField(sortField: string){
+        if (this._sortField !== sortField) {
+            this.links = {};
+            this._sortField = sortField;
+        }
+    }
+
+    get sortOrder(): number{
+        return this._sortOrder;
+    }
+
+    set sortOrder(sortOrder: number){
+        if (this._sortOrder !== sortOrder) {
+            this.links = {};
+            this._sortOrder = sortOrder;
+        }
+    }
+
+    /**
+     * Send a pagination request with url as base URL, if url is null or undefined then
+     * it use the url property value instead.
+     * Also it use the values of sortField, sortOrder and filter properties like query parameters,
+     * so the finally url will be:
+     * <url>/orderby=[sortField-value]&direction=[sortOrder-value]&filter=[filter-value]
+     * @param url base url
+     */
+    public get(url?: string): Observable<any[]> {
+        let params: URLSearchParams = new URLSearchParams();
+
+        if (this.sortField) {
+            params.set('orderby', this.sortField);
+        }
+
+        if (this.sortOrder) {
+            params.set('direction', this.sortOrder === 1 ?
+                PaginatorService.OrderDirection.ASC : PaginatorService.OrderDirection.DESC);
+        }
+
+        if (this.filter) {
+            params.set('filter', `${this.filter}`);
+        }
+
+        return this.coreWebService.requestView({
+            method: RequestMethod.Get,
+            search: params,
+            url: !url ? this.url : url
+        }).map(response => {
+            this.setLinks(response.header(PaginatorService.LINK_HEADER_NAME));
+            this.paginationPerPage = parseInt(response.header(PaginatorService.PAGINATION_PER_PAGE_HEADER_NAME), 10);
+            this.currentPage = parseInt(response.header(PaginatorService.PAGINATION_CURRENT_PAGE_HEADER_NAME), 10);
+            this.maxLinksPage = parseInt(response.header(PaginatorService.PAGINATION_MAX_LINK_PAGES_HEADER_NAME), 10);
+            this.totalRecords = parseInt(response.header(PaginatorService.PAGINATION_TOTAL_ENTRIES_HEADER_NAME), 10);
+            return response.entity;
+        });
+    }
+
+    /**
+     * Request the last page
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorService
+     */
+    public getLastPage(): Observable<any[]> {
+        return this.get(this.links.last);
+    }
+
+    /**
+     * Request the first page
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorService
+     */
+    public getFirstPage(): Observable<any[]> {
+        return this.get(this.links.first);
+    }
+
+    /**
+     * request the  pageParam page
+     * @param {number} [pageParam=1] Page to request
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorServic
+     */
+    public getPage(pageParam = 1): Observable<any[]> {
```

Comment:
In Typescript when you assign a value it will assign the type by default... in fact if you add the type tslint will tell you it's redundant... at least that's what happen in a variable or property, not sure in a param like this... but I guess it's the same thing.